ipam/allocator.go

```diff
@@ -51,9 +52,11 @@ type Allocator struct {
 	nicknames        map[router.PeerName]string   // so we can map nicknames for rmpeer
 	pendingAllocates []operation                  // held until we get some free space
 	pendingClaims    []operation                  // held until we know who owns the space
+	dead             map[string]time.Time         // containers we heard were dead, and when
```

Comment:
Thank you for the suggestions. 

`dead` is used in just 4 lines at present, so I think the code expansion to implement a heap would be overwhelming. 

`removeDeadContainers` is called once per 5 seconds, and over time there will be a balance between the number of containers it skips over and the number it needs to clear down (which is the number it skipped over last time).  Given that the actual clear-down is more work than the skipping, I do not think it can be made significantly cheaper overall.